README.md

```diff
@@ -1,58 +1,151 @@
+- Metadata includes:
+  - File name
+  - Identifier
+  - Description
+  - Format
+  - License
+  - Size
+  - Hash
+  - Issued date
+  - Modified date
+  - Duration
+  - Topics and message counts
+
+### Usage
+
+#### 1. Setup
+
+Ensure all dependencies are installed. You can use the following command to install required packages:
+
+```bash
+pip install mcap
+```
+
+#### 2. Running the Script
+
+To generate the metadata JSON file, follow these steps:
+
+- Place all your MCAP files in a directory.
+- The default directory is `/recorded_datasets/edinburgh`
+- Run the script:
+
+  ```bash
+  python metadata_generator.py
+  ```
+
+If you want to generate metadata for files in a specified path, run the script:
+
+```bash
+python metadata_generator.py -p path/to/file
+```
+
+#### 3. Output
+
+The script will generate a `resources.json` file in the specified directory. This JSON file will contain metadata for each MCAP file in the directory.
+
+## Upload_vehicle_data usage
+This script automates the upload of rosbags from a vehicle to a cloud infrastructure, specifically within the EIDF (Edinburgh International Data Facility) cloud instance. The primary purpose of this script is to streamline the process of data collection and transfer, ensuring efficient handling and storage of large datasets generated by vehicle sensors.
+
+- EIDF Cloud Instance: The EIDF cloud provides robust and scalable infrastructure for data storage and computation, which is crucial for handling the large volumes of data generated by autonomous vehicles. By utilizing the EIDF cloud, the script ensures that data is securely stored and readily available for further analysis and processing.
+
+- Data Collection: The rosbags collected contain crucial data from various sensors on the vehicle, including LiDAR, cameras, and GPS. This data is essential for research and development in autonomous driving, enabling tasks such as sensor fusion, environment mapping, and behavior prediction.
+
+- Automation and Efficiency: Manually uploading large datasets can be time-consuming and prone to errors. This script automates the entire process, from compressing the rosbags to transferring them securely to the remote host machine. It also includes functionality to manage disk space efficiently on the remote machine.
+
+### 1. Dependencies
+
+
+  ##### 1. Remote host machine
+- iperf3: A tool for measuring bandwidth.
+```bash
+sudo apt-get install iperf3
+```
+- set SSH keys authentication: This should be done before using this script.
+   - create an ssh-key:
+   ```bash
+   ssh-keygen -t rsa -b 2048
+   ```
+   - share a SSH Key
+   ```bash
+   ssh-copy-id username@hostname_ip
+   ```
+
+#### 2. Remote machine
+- iperf3: A tool for measuring bandwidth.
+```bash
+sudo apt-get install iperf3
+```
+- MCAP cli: A tool for compressing rosbags
+```bash
+wget -O /home/mcap https://github.com/foxglove/mcap/releases/download/releases%2Fmcap-cli%2Fv0.0.47/mcap-linux-amd64 && chmod +x $HOME/mcap

```

Comment:
Revert this to

```suggestion
wget -O $HOME/mcap https://github.com/foxglove/mcap/releases/download/releases%2Fmcap-cli%2Fv0.0.47/mcap-linux-amd64 && chmod +x $HOME/mcap

```
 Writing directly into `/home`  will require `sudo` so lets just have a cmd that anyone can run

Reply:
Changed